File: src/main/java/com/example/model/Product.java
package com.example.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.validation.constraints.NotNull;

@Entity
@Table(name="tb_product")
public class Product {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name="product_id")
	private int productId;
	
	private String productCode;
	private String productName;
	private String productDesc;
	
	//@Temporal(TemporalType.TIMESTAMP)
	private Date regDate = new Date();
	
	private String regUserid;
	
	public int getProductId() {
		return productId;
	}
	public void setProductid(int productId) {
		this.productId = productId;
	}
	public String getProductCode() {
		return productCode;
	}
	public void setProductCode(String productCode) {
		this.productCode = productCode;
	}
	public String getProductName() {
		return productName;
	}
	public void setProductName(String productName) {
		this.productName = productName;
	}
	public String getProductDesc() {
		return productDesc;
	}
	public void setProductDesc(String productDesc) {
		this.productDesc = productDesc;
	}
	public Date getRegDate() {
		return regDate;
	}
	public void setRegDate(Date regDate) {
		this.regDate = regDate;
	}
	public String getRegUserid() {
		return regUserid;
	}
	public void setRegUserid(String regUserid) {
		this.regUserid = regUserid;
	}
}

File: src/main/webapp/app/product/product.ts
import * as _ from 'lodash'

export class Product {
    productId:number;
    productCode:string;
    productName:string; 
    productDesc:String;
    regDate:String;
	regUserid:String;

     constructor(product?:{productId:number, productCode:string, productName:string, productDesc:String,
    regDate:String, regUserid:String}) {
        if(product) {
            _.assignIn(this, product);
        }
        //this.productId = product.productId;
        // this.productCode = product.productCode;
        // this.productName = product.productName;
        // this.productDesc = product.productDesc;
        // this.regDate = product.regDate;
        // this.regUserid = product.regUserid;
     }
}